src/bridge/setup.rs

```diff
@@ -142,9 +173,33 @@ pub async fn setup_neovide_specific_state(
         .ok();
 
     // Create auto command for retrieving exit code from neovim on quit.
-    nvim.command("autocmd VimLeave * call rpcnotify(1, 'neovide.quit', v:exiting)")
-        .await
-        .ok();
+    nvim.command(
+        "autocmd VimLeave * call rpcnotify(g:neovide_channel_id, 'neovide.quit', v:exiting)",
+    )
+    .await
+    .ok();
+
+    nvim.command(
+        "autocmd OptionSet columns call rpcnotify(g:neovide_channel_id, 'neovide.columns', str2nr(v:option_new))",
+    )
+    .await
+    .ok();
+    nvim.command(
+        "autocmd OptionSet lines call rpcnotify(g:neovide_channel_id, 'neovide.lines', str2nr(v:option_new))",
+    )
+    .await
+    .ok();
```

Comment:
I will create a Neovide feature request for sending `columns` and `lines` as UI options.  So that this wouldn't be needed. 

Another thing to note here is that these don't trigger on automatic resizes with `nvim_try_resize`, which is ideal for avoiding strange race conditions. The user is unlikely to both set those and resize the window manually at the same time.